### PR TITLE
refactor: format files with uncrustify

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Review by justinmk — APPROVED** [review]

File: src/nvim/ex_docmd.c
Comment (bfredl) [inline src/nvim/ex_docmd.c:2119]: this one a little weird. probably uncrustify failed to consider `if (expr) {` on the same line as `case 'c'` (and the fix is to move the `if` to the line below the `case` ).

Comment (bfredl) [reply src/nvim/ex_docmd.c:2119]: there's a few more cases like this, could be fixed manually with a suitable pattern for now :)

Comment (dundargoc) [reply src/nvim/ex_docmd.c:2119]: It's possible to add a newline after a case. Want me to try it out and add it as a separate commit so you can check if it looks good?

Comment (bfredl) [reply src/nvim/ex_docmd.c:2119]: yes.

Comment (dundargoc) [reply src/nvim/ex_docmd.c]: Added newline after case, but only if there's no brace after.

Comment (dundargoc) [reply src/nvim/ex_docmd.c]: While you're at it, could you take an extra look at the indentation on switch/case? I think there's currently no indentation between switch and case, is that correct?

Comment (justinmk) [reply src/nvim/ex_docmd.c]: > currently no indentation between switch and case, is that correct?

correct, though I don't think it matters much. Except that the current rule avoids some gratuitous indentation.

File: contrib/uncrustify.cfg
Comment (justinmk) [inline contrib/uncrustify.cfg:1646]: +1

**Review by justinmk — APPROVED** [review]

File: src/nvim/ops.c
Comment (bfredl) [inline src/nvim/ops.c:3482]: something went wrong here? :)

File: src/nvim/ex_docmd.c
Comment (bfredl) [inline src/nvim/ex_docmd.c:5105]: I think you can add a `@return` here (I don't think there's any requirement to put `@return` below params, that's just a convention)

File: src/nvim/ex_docmd.c
Comment (bfredl) [inline src/nvim/ex_docmd.c:5847]: same

File: src/nvim/ex_docmd.c
Comment (bfredl) [inline src/nvim/ex_docmd.c:8932]: same

**Comment by bfredl** [comment]
LGTM after final comments :)

File: src/nvim/ex_docmd.c
Comment (dundargoc) [reply src/nvim/ex_docmd.c:5105]: :+1:

File: src/nvim/ops.c
Comment (dundargoc) [reply src/nvim/ops.c]: I've added this fix as a separate commit. Take a look and see if it looks alright.

File: src/nvim/ops.c
Comment (bfredl) [inline src/nvim/ops.c:2298]: this doesn't look right, if there _needs_ to be a linebreak, the comparisation operation should go on the new line, also ideally it should be intended within the `&&` .

File: src/nvim/ops.c
Comment (dundargoc) [reply src/nvim/ops.c]: Uncrustify is better at _not_ splitting lines than joining them. I think the simplest fix would be to adjust this manually and uncrustify won't touch it afterwards. Is that alright solution?

Comment (bfredl) [reply src/nvim/ops.c:3482]: Sure let's be conservative and do it manually if uncrustify cannot be taught the right thing

Comment (dundargoc) [reply src/nvim/ops.c]: :+1:

**Comment by dundargoc** [comment]
@bfredl @justinmk Ok, done with all review comments.

**Comment by justinmk** [comment]
fyi for future PRs with big changes, it is very helpful to use "fixup" commits so that it is easy to see what changed

**Review by justinmk — APPROVED** [review]

**Comment by bfredl** [comment]
@justinmk to be fair he did, and squashed after I approved of the fixup changes.

**Comment by bfredl** [comment]
merged. there's a few funky lines left (around stuff like really long string constants) but let's not stall the **massive** reduction of `clint` noise that is in the balance.

**Comment by justinmk** [comment]
Might be better to squash-merge PRs like this, to avoid needing both the merge-commit and the actual commit in the .git-blame-ignore-revs

**Comment by dundargoc** [comment]
> fyi for future PRs with big changes, it is very helpful to use "fixup" commits so that it is easy to see what changed

Like @bfredl mentioned, I made a separate commit for each comment where I showed all the changes. After the changes were confirmed to be OK I squashed it to avoid excessive commits. Would you like me to skip squashing the fixup commits?

**Comment by Zeeshan376** [comment]
Plzz not sent email plzzzx

**Comment by clason** [comment]
> Plzz not sent email plzzzx
> […](#)

Don't watch this PR (or the Neovim repo in general) then.

**Comment by justinmk** [comment]
> Like @bfredl mentioned, I made a separate commit for each comment where I showed all the changes. 

@dundargoc yeah that was my mistake

> After the changes were confirmed to be OK I squashed it to avoid excessive commits. Would you like me to skip squashing the fixup commits?

Not necessary unless you have intentionally separated commits which should be merged instead of squash-merged. Otherwise, whoever merges the PR should use the squash-merge option.

**Pull request merged**
